pkg/re_exec/interface.go

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright (C) 2024 IOTech Ltd
+//
+
+package re_exec
+
+import (
+	"github.com/IOTechSystems/go-mod-edge-utils/pkg/errors"
+)
+
+// ReExecQueue is the interface for the retry queue
+type ReExecQueue[T any] interface {
```

Comment:
move this interface to pkg.common, and rename to `Queue` because I just realized there is no ReExec func in this interface.  It may be applied for other purposes in the future.